dev/TabView/TabView.cpp

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "TabView.h"
+#include "TabViewAutomationPeer.h"
+#include "DoubleUtil.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+#include "SharedHelpers.h"
+
+static constexpr double c_tabMinimumWidth = 48.0;
+static constexpr double c_tabMaximumWidth = 200.0;
+
+static constexpr wstring_view c_tabViewItemMinWidthName{ L"TabViewItemMinWidth"sv };
+static constexpr wstring_view c_tabViewItemMaxWidthName{ L"TabViewItemMaxWidth"sv };
+
+// TODO: what is the right number and should this be customizable?
+static constexpr double c_scrollAmount = 50.0;
+
+TabView::TabView()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_TabView);
+
+    SetDefaultStyleKey(this);
+
+    Loaded({ this, &TabView::OnLoaded });
+    SelectionChanged({ this, &TabView::OnSelectionChanged });
+    SizeChanged({ this, &TabView::OnSizeChanged });
+}
+
+void TabView::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    m_tabContentPresenter.set(GetTemplateChildT<winrt::ContentPresenter>(L"TabContentPresenter", controlProtected));
+    m_scrollViewer.set(GetTemplateChildT<winrt::FxScrollViewer>(L"ScrollViewer", controlProtected));
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        m_scrollViewerLoadedRevoker = scrollViewer.Loaded(winrt::auto_revoke, { this, &TabView::OnScrollViewerLoaded });
+    }
+}
+
+void TabView::OnTabWidthModePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
```

Comment:
>args [](start = 93, length = 4)

unused parameters should be unnamed, delete args here

Reply:
ok